test/powershell/Modules/Microsoft.PowerShell.Utility/Sort-Object.Tests.ps1

```diff
@@ -21,204 +21,946 @@
+        $results = $employees | Sort-Object -Property @($ht1,$ht2) -Descending
+        
+        $results[0] | Should Be $employees[2]
+        $results[1] | Should Be $employees[1]
+        $results[2] | Should Be $employees[0]
+    }
+
+    It "Sort-Object with -Descending:$false and Two Order Entry Keys should work"{
+        $employee1 = [pscustomobject]@{"FirstName"="john"; "LastName"="smith"; "YearsInMS"=5}
+        $employee2 = [pscustomobject]@{"FirstName"="joesph"; "LastName"="smith"; "YearsInMS"=15}
+        $employee3 = [pscustomobject]@{"FirstName"="john"; "LastName"="smyth"; "YearsInMS"=12}
+        $employees = @($employee1,$employee2,$employee3)
+        $results = $employees | Sort-Object -Property "LastName","FirstName" -Descending:$false
+        
+        $results[0] | Should Be $employees[1]
+        $results[1] | Should Be $employees[0]
+        $results[2] | Should Be $employees[2]
+    }
+
+    It "Sort-Object with -Descending and Two Order Entry Keys should work"{
+        $employee1 = [pscustomobject]@{"FirstName"="john"; "LastName"="smith"; "YearsInMS"=5}
+        $employee2 = [pscustomobject]@{"FirstName"="joesph"; "LastName"="smith"; "YearsInMS"=15}
+        $employee3 = [pscustomobject]@{"FirstName"="john"; "LastName"="smyth"; "YearsInMS"=12}
+        $employees = @($employee1,$employee2,$employee3)
+        $results = $employees | Sort-Object -Property "LastName","FirstName" -Descending
+        
+        $results[0] | Should Be $employees[2]
+        $results[1] | Should Be $employees[0]
+        $results[2] | Should Be $employees[1]
+    }
+
+    It "Sort-Object with Two Order Entry Keys with asc=true should work"{
+        $employee1 = [pscustomobject]@{"FirstName"="john"; "LastName"="smith"; "YearsInMS"=5}
+        $employee2 = [pscustomobject]@{"FirstName"="joesph"; "LastName"="smith"; "YearsInMS"=15}
+        $employee3 = [pscustomobject]@{"FirstName"="john"; "LastName"="smyth"; "YearsInMS"=12}
+        $employees = @($employee1,$employee2,$employee3)
+        $ht1 = @{"e"="FirstName"; "asc"=$true}
+        $ht2 = @{"e"="LastName";}
+        $results = $employees | Sort-Object -Property @($ht1,$ht2) -Descending
+        
+        $results[0] | Should Be $employees[1]
+        $results[1] | Should Be $employees[2]
+        $results[2] | Should Be $employees[0]
+    }
+
+    It "Sort-Object with Descending No Property should work"{
+        $employee1 = 1
+        $employee2 = 2
+        $employee3 = 3
+        $employees = @($employee1,$employee2,$employee3)
+        $results = $employees | Sort-Object -Descending
+        
+        $results[0] | Should Be 3
+        $results[1] | Should Be 2
+        $results[2] | Should Be 1
+    }
 }
+
+Describe 'Sort-Object Top and Bottom Unit Tests' -Tags 'CI' {
+
+    It 'Return the top N integers sorted in ascending order' {
+        $unsortedData = 973474993,271612178,-1258909473,659770354,1829227828,-1709391247,-10835210,-1477737798,1125017828,813732193
```

Comment:
BeforeAll isn't documented (I didn't even know it existed), and I simply followed the model in the file already.

The numeric data used was intentionally generated using Get-Random with the range of all int values so that the test would result in a random outcome that covered a broad base of numeric input.  With sorting and -Top/-Bottom, 5 or 6 values to start with is not always a sufficient test, so I'll leave it with 10 values.